Update git.md with more details about squash/amend commits (#2617)
google/android-fhir: docs/contrib/git.md

This project generally follows [these guidelines](https://github.com/google/fhir-data-pipes/blob/master/doc/review_process.md) from our `fhir-data-pipes` sister project.

## Overview

We use the following workflow:

> 1. Contributors can open "draft" PRs for any Work In Progress (WIP) which does not require reviews yet.

Reviewer: ```suggestion
1. Contributors can open "draft" PRs for any Work In Progress (WIP) which does not require review yet.
```

>    It's OK if these don't fully build, yet.

Reviewer: ```suggestion
   It's OK if the checks do not pass yet.
```

1. When a PR is ready for code review, Contributors un-draft it. It must fully build green.

>    At this point, PRs should only have 1 single commit; please "squash" (or use `--amend`) your "local history",

Reviewer: ```suggestion
   At this point, the PR should only have 1 single commit; please "squash" (or use `--amend`) your "local history",
```

   before pushing to a branch to open a PR (or when your WIP Draft PR is ready for review). The commit message
   of this initial commit should explain what this PR is all about.

1. Maintainers, or other Contributors, will now review the PR. They may add comments requesting changes.

1. When maintainers update PRs to make changes requested by reviewers, those should be added as
   additional new single commits per round of review,
   typically with a generic message such as _"Incorporated review feedback."_
   Do NOT squash (or amend) these review updates into the original commit.

> 1. Maintainers (AKA Committers) _squash_ all commits of PRs into a single commit when merging.

Reviewer 2: Just FYI, in the backend repos (i.e., data-pipes and gateway) we actually disable other options to make sure that all merges are done with squash. You can do that under Settings > General:
![image](https://github.com/user-attachments/assets/fe3b3dd6-8748-47f6-bcc2-79efc53a75c3)

Reviewer: we have the same :)
